ls1intum/Artemis
# New component submission result status

### Checklist

### Motivation and Context
The idea for this PR originated as part of [this](https://github.com/ls1intum/Artemis/pull/1167#issuecomment-572970796) discussion. The goal is to remove the code duplication for displaying the submission result status found in `course-exercise-details.component.html` and `course-exercise-row.component.html`, and to make the result string consistent among the different pages (see linked comment before for more details).

### Description
- Create new component `submission-result-status`
- Use new component in `course-exercise-details.component.html` and `course-exercise-row.component.html` (with respective input values to account for the differences)
- Use new component in `text-editor`, `modeling-submission` and `file-upload-submission` (this will lead to consistent submission result status texts, up until now on the submission pages themselves a slightly different status text could be shown, e.g. `No graded result` when it showed `No submission` in other places (on course overview and exercise detail).

### Steps for Testing
1. Log in to Artemis
2. Navigate to Overview
3. Click on a Course
4. Check that the submission result status looks as before (short version, no badge)
5. Click on different exercises and check that the submission result status looks as before (long version, with badge)
6. Go to submission pages for Text, Modeling and File Upload and check that the result status text is exactly the same as on the exercise detail page (without the badge though).

### Screenshots
Before/after is the same (focus in on submission result status):
![OnPaste 20200112-134901](https://user-images.githubusercontent.com/7109225/72219120-b17b8f80-3542-11ea-835e-df78ee547076.png)

Before/after is the same (focus in on submission result status):
![OnPaste](https://user-images.githubusercontent.com/7109225/72219127-cd7f3100-3542-11ea-8719-fb958836a575.png)

This has changed for certain cases, e.g. was previously saying `No graded result` instead of `No submission`:
![OnPaste 20200120-192801](https://user-images.githubusercontent.com/7109225/72749673-fc3c7d80-3bba-11ea-8a11-5cf030dbf054.png)
